build-disk panics if no config provided

```
$ sudo bin/elemental build-disk --debug                            
DEBU[2022-06-09T14:22:43+02:00] Starting elemental version v0.0.15 on commit 73aa161d1a5d82a2117e150030d60f1a16830824 
DEBU[2022-06-09T14:22:43+02:00] Full config loaded: &v1.BuildConfig{
  Date: false,
  Name: "elemental",
  OutDir: "",
  Config: v1.Config{
    Logger: &v1.logrusWrapper{ // p0
      Logger: &logrus.Logger{
        Out: &os.File{},
        Hooks: logrus.LevelHooks{},
        Formatter: &logrus.TextFormatter{
          ForceColors: true,
          DisableColors: false,
          ForceQuote: false,
          DisableQuote: false,
          EnvironmentOverrideColors: false,
          DisableTimestamp: false,
          FullTimestamp: true,
          TimestampFormat: "",
          DisableSorting: false,
          SortingFunc: ,
          DisableLevelTruncation: false,
          PadLevelText: false,
          QuoteEmptyFields: false,
          FieldMap: logrus.FieldMap(nil),
          CallerPrettyfier: ,
        },
        ReportCaller: false,
        Level: 5,
        ExitFunc: os.Exit,
      },
    },
    Fs: &vfs.osfs{}, // p1
    Mounter: &mount.FakeMounter{
      MountPoints: nil,
      MountCheckErrors: map[string]error(nil),
      UnmountFunc: ,
    },
    Runner: &v1.RealRunner{ // p2
      Logger: p0,
    },
    Syscall: &v1.RealSyscall{},
    CloudInitRunner: &cloudinit.YipCloudInitRunner{},
    Luet: &luet.Luet{
      VerifyImageUnpack: false,
      TmpDir: "",
    },
    Client: &http.Client{},
    Cosign: false,
    Verify: false,
    CosignPubKey: "",
    LocalImage: false,
    Repos: []v1.Repository{
      v1.Repository{
        Name: "cos",
        Priority: 90,
        URI: "quay.io/costoolkit/releases-green",
        Type: "docker",
        Arch: "",
        ReferenceID: "",
      },
    },
    Arch: "x86_64",
    SquashFsCompressionConfig: []string{
      "-comp",
      "xz",
      "-Xbcj",
      "x86",
    },
  },
} 
WARN[2022-06-09T14:22:43+02:00] error unmarshalling RawDisk: 1 error(s) decoding:

* '[debug]' expected a map, got 'bool' 
DEBU[2022-06-09T14:22:43+02:00] Loaded RawDisk: &v1.RawDisk{}                
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11a7e4e]

goroutine 1 [running]:
github.com/rancher/elemental-cli/cmd.NewBuildDisk.func2(0xc00041bb80, {0xc000287550?, 0x1?, 0x1?})
	/home/itxaka/projects/elemental-cli/cmd/build-disk.go:79 +0x1ce
github.com/spf13/cobra.(*Command).execute(0xc00041bb80, {0xc000287540, 0x1, 0x1})
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00041b900)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/rancher/elemental-cli/cmd.Execute()
	/home/itxaka/projects/elemental-cli/cmd/root.go:48 +0x25
main.main()
	/home/itxaka/projects/elemental-cli/main.go:21 +0x17

```

@frelon is you binary rebuilt? Im on the same branch, latest main and still getting it!

`commit 73aa161d1a5d82a2117e150030d60f1a16830824 HEAD (HEAD -> main, upstream/main, origin/main)`

```
$ make build && sudo bin/elemental build-disk --debug                            
go build -ldflags '-w -s -X "github.com/rancher/elemental-cli/internal/version.version=v0.0.15" -X "github.com/rancher/elemental-cli/internal/version.gitCommit=73aa161d1a5d82a2117e150030d60f1a16830824"' -o bin/elemental
DEBU[2022-06-09T14:40:47+02:00] Starting elemental version v0.0.15 on commit 73aa161d1a5d82a2117e150030d60f1a16830824 
DEBU[2022-06-09T14:40:47+02:00] Full config loaded: &v1.BuildConfig{
  Date: false,
  Name: "elemental",
  OutDir: "",
  Config: v1.Config{
    Logger: &v1.logrusWrapper{ // p0
      Logger: &logrus.Logger{
        Out: &os.File{},
        Hooks: logrus.LevelHooks{},
        Formatter: &logrus.TextFormatter{
          ForceColors: true,
          DisableColors: false,
          ForceQuote: false,
          DisableQuote: false,
          EnvironmentOverrideColors: false,
          DisableTimestamp: false,
          FullTimestamp: true,
          TimestampFormat: "",
          DisableSorting: false,
          SortingFunc: ,
          DisableLevelTruncation: false,
          PadLevelText: false,
          QuoteEmptyFields: false,
          FieldMap: logrus.FieldMap(nil),
          CallerPrettyfier: ,
        },
        ReportCaller: false,
        Level: 5,
        ExitFunc: os.Exit,
      },
    },
    Fs: &vfs.osfs{}, // p1
    Mounter: &mount.FakeMounter{
      MountPoints: nil,
      MountCheckErrors: map[string]error(nil),
      UnmountFunc: ,
    },
    Runner: &v1.RealRunner{ // p2
      Logger: p0,
    },
    Syscall: &v1.RealSyscall{},
    CloudInitRunner: &cloudinit.YipCloudInitRunner{},
    Luet: &luet.Luet{
      VerifyImageUnpack: false,
      TmpDir: "",
    },
    Client: &http.Client{},
    Cosign: false,
    Verify: false,
    CosignPubKey: "",
    LocalImage: false,
    Repos: []v1.Repository{
      v1.Repository{
        Name: "cos",
        Priority: 90,
        URI: "quay.io/costoolkit/releases-green",
        Type: "docker",
        Arch: "",
        ReferenceID: "",
      },
    },
    Arch: "x86_64",
    SquashFsCompressionConfig: []string{
      "-comp",
      "xz",
      "-Xbcj",
      "x86",
    },
  },
} 
WARN[2022-06-09T14:40:47+02:00] error unmarshalling RawDisk: 1 error(s) decoding:

* '[debug]' expected a map, got 'bool' 
DEBU[2022-06-09T14:40:47+02:00] Loaded RawDisk: &v1.RawDisk{}                
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11a7e4e]

goroutine 1 [running]:
github.com/rancher/elemental-cli/cmd.NewBuildDisk.func2(0xc000308f00, {0xc00063f380?, 0x1?, 0x1?})
	/home/itxaka/projects/elemental-cli/cmd/build-disk.go:79 +0x1ce
github.com/spf13/cobra.(*Command).execute(0xc000308f00, {0xc00063f370, 0x1, 0x1})
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000308c80)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/rancher/elemental-cli/cmd.Execute()
	/home/itxaka/projects/elemental-cli/cmd/root.go:48 +0x25
main.main()
	/home/itxaka/projects/elemental-cli/main.go:21 +0x17

# itxaka @ zeus in ~/projects/elemental-cli on git:main o [14:40:47] C:2
$ make build && sudo bin/elemental version --long && sudo bin/elemental build-disk --debug
go build -ldflags '-w -s -X "github.com/rancher/elemental-cli/internal/version.version=v0.0.15" -X "github.com/rancher/elemental-cli/internal/version.gitCommit=73aa161d1a5d82a2117e150030d60f1a16830824"' -o bin/elemental
version.BuildInfo{Version:"v0.0.15", GitCommit:"73aa161d1a5d82a2117e150030d60f1a16830824", GoVersion:"go1.18.2"}DEBU[2022-06-09T14:41:05+02:00] Starting elemental version v0.0.15 on commit 73aa161d1a5d82a2117e150030d60f1a16830824 
DEBU[2022-06-09T14:41:05+02:00] Full config loaded: &v1.BuildConfig{
  Date: false,
  Name: "elemental",
  OutDir: "",
  Config: v1.Config{
    Logger: &v1.logrusWrapper{ // p0
      Logger: &logrus.Logger{
        Out: &os.File{},
        Hooks: logrus.LevelHooks{},
        Formatter: &logrus.TextFormatter{
          ForceColors: true,
          DisableColors: false,
          ForceQuote: false,
          DisableQuote: false,
          EnvironmentOverrideColors: false,
          DisableTimestamp: false,
          FullTimestamp: true,
          TimestampFormat: "",
          DisableSorting: false,
          SortingFunc: ,
          DisableLevelTruncation: false,
          PadLevelText: false,
          QuoteEmptyFields: false,
          FieldMap: logrus.FieldMap(nil),
          CallerPrettyfier: ,
        },
        ReportCaller: false,
        Level: 5,
        ExitFunc: os.Exit,
      },
    },
    Fs: &vfs.osfs{}, // p1
    Mounter: &mount.FakeMounter{
      MountPoints: nil,
      MountCheckErrors: map[string]error(nil),
      UnmountFunc: ,
    },
    Runner: &v1.RealRunner{ // p2
      Logger: p0,
    },
    Syscall: &v1.RealSyscall{},
    CloudInitRunner: &cloudinit.YipCloudInitRunner{},
    Luet: &luet.Luet{
      VerifyImageUnpack: false,
      TmpDir: "",
    },
    Client: &http.Client{},
    Cosign: false,
    Verify: false,
    CosignPubKey: "",
    LocalImage: false,
    Repos: []v1.Repository{
      v1.Repository{
        Name: "cos",
        Priority: 90,
        URI: "quay.io/costoolkit/releases-green",
        Type: "docker",
        Arch: "",
        ReferenceID: "",
      },
    },
    Arch: "x86_64",
    SquashFsCompressionConfig: []string{
      "-comp",
      "xz",
      "-Xbcj",
      "x86",
    },
  },
} 
WARN[2022-06-09T14:41:05+02:00] error unmarshalling RawDisk: 1 error(s) decoding:

* '[debug]' expected a map, got 'bool' 
DEBU[2022-06-09T14:41:05+02:00] Loaded RawDisk: &v1.RawDisk{}                
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11a7e4e]

goroutine 1 [running]:
github.com/rancher/elemental-cli/cmd.NewBuildDisk.func2(0xc0003ad680, {0xc000254790?, 0x1?, 0x1?})
	/home/itxaka/projects/elemental-cli/cmd/build-disk.go:79 +0x1ce
github.com/spf13/cobra.(*Command).execute(0xc0003ad680, {0xc000254780, 0x1, 0x1})
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003ad400)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/itxaka/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/rancher/elemental-cli/cmd.Execute()
	/home/itxaka/projects/elemental-cli/cmd/root.go:48 +0x25
main.main()
	/home/itxaka/projects/elemental-cli/main.go:21 +0x17

```

--- a/cmd/build-disk.go
+++ b/cmd/build-disk.go
@@ -67,6 +67,14 @@ func NewBuildDisk(root *cobra.Command, addCheckRoot bool) *cobra.Command {
 				return err
 			}
 
+			// Get the spec for the arch only
+			var specArch *v1.RawDiskArchEntry
+			if cfg.Arch == "x86_64" {
+				specArch = spec.X86_64
+			} else {
+				specArch = spec.Arm64
+			}
+
 			// TODO map these to buildconfig and rawdisk structs, so they
 			// are directly unmarshaled and there is no need handle them here
 			imgType, _ := flags.GetString("type")
@@ -76,7 +84,7 @@ func NewBuildDisk(root *cobra.Command, addCheckRoot bool) *cobra.Command {
 
 			// Set the repo depending on the arch we are building for
 			var repos []v1.Repository
-			for _, u := range (*spec)[cfg.Arch].Repositories {
+			for _, u := range specArch.Repositories {
 				repos = append(repos, v1.Repository{
 					URI:         u.URI,
 					Priority:    constants.LuetDefaultRepoPrio,
@@ -97,7 +105,7 @@ func NewBuildDisk(root *cobra.Command, addCheckRoot bool) *cobra.Command {
 				return fmt.Errorf("output file %s exists, refusing to continue", output)
 			}
 
-			err = action.BuildDiskRun(cfg, spec, imgType, oemLabel, recoveryLabel, output)
+			err = action.BuildDiskRun(cfg, specArch, imgType, oemLabel, recoveryLabel, output)
 			if err != nil {
 				return err
 			}

--- a/cmd/config/config.go
+++ b/cmd/config/config.go
@@ -283,7 +283,7 @@ func ReadBuildISO(b *v1.BuildConfig, flags *pflag.FlagSet) (*v1.LiveISO, error) 
 }
 
 func ReadBuildDisk(b *v1.BuildConfig, flags *pflag.FlagSet) (*v1.RawDisk, error) {
-	disk := config.NewRawDisk(b.Config)
+	disk := config.NewRawDisk()
 	vp := viper.Sub("raw_disk")
 	if vp == nil {
 		vp = viper.New()

--- a/cmd/config/config_test.go
+++ b/cmd/config/config_test.go
@@ -164,10 +164,10 @@ var _ = Describe("Config", Label("config"), func() {
 				Expect(err).ShouldNot(HaveOccurred())
 
 				// From config file
-				Expect(len((*disk)["x86_64"].Packages)).To(Equal(1))
-				Expect((*disk)["x86_64"].Packages[0].Name).To(Equal("system/myos"))
-				Expect(len((*disk)["x86_64"].Repositories)).To(Equal(1))
-				Expect((*disk)["x86_64"].Repositories[0].URI).To(Equal("quay.io/some/repo"))
+				Expect(len(disk.X86_64.Packages)).To(Equal(1))
+				Expect(disk.X86_64.Packages[0].Name).To(Equal("system/myos"))
+				Expect(len(disk.X86_64.Repositories)).To(Equal(1))
+				Expect(disk.X86_64.Repositories[0].URI).To(Equal("quay.io/some/repo"))
 			})
 		})
 	})

--- a/pkg/action/build-disk.go
+++ b/pkg/action/build-disk.go
@@ -35,15 +35,10 @@ import (
 var MB = int64(1024 * 1024)
 var GB = 1024 * MB
 
-func BuildDiskRun(cfg *v1.BuildConfig, spec *v1.RawDisk, imgType string, oemLabel string, recoveryLabel string, output string) (err error) {
+func BuildDiskRun(cfg *v1.BuildConfig, spec *v1.RawDiskArchEntry, imgType string, oemLabel string, recoveryLabel string, output string) (err error) {
 	cfg.Logger.Infof("Building disk image type %s for arch %s", imgType, cfg.Arch)
-	if (*spec)[cfg.Arch] == nil {
-		msg := fmt.Sprintf("no values in the config for arch %s", cfg.Arch)
-		cfg.Logger.Error(msg)
-		return errors.New(msg)
-	}
 
-	if len((*spec)[cfg.Arch].Packages) == 0 {
+	if len(spec.Packages) == 0 {
 		msg := fmt.Sprintf("no packages in the config for arch %s", cfg.Arch)
 		cfg.Logger.Error(msg)
 		return errors.New(msg)
@@ -84,15 +79,16 @@ func BuildDiskRun(cfg *v1.BuildConfig, spec *v1.RawDisk, imgType string, oemLabe
 	rootfsPart := filepath.Join(diskTempDir, "rootfs.part")
 	oemPart := filepath.Join(diskTempDir, "oem.part")
 	efiPart := filepath.Join(diskTempDir, "efi.part")
-
 	// Extract required packages to basedir
-	for _, pkg := range (*spec)[cfg.Arch].Packages {
+	for _, pkg := range spec.Packages {
 		err = os.MkdirAll(filepath.Join(baseDir, pkg.Target), constants.DirPerm)
 		if err != nil {
+			cfg.Logger.Error(err)
 			return err
 		}
 		imgSource, err := v1.NewSrcFromURI(pkg.Name)
 		if err != nil {
+			cfg.Logger.Error(err)
 			return err
 		}
 		err = e.DumpSource(

--- a/pkg/action/build_test.go
+++ b/pkg/action/build_test.go
@@ -232,8 +232,8 @@ var _ = Describe("Runtime Actions", func() {
 	Describe("Build disk", Label("disk", "build"), func() {
 		var rawDisk *v1.RawDisk
 		BeforeEach(func() {
-			rawDisk = config.NewRawDisk(cfg.Config)
-			(*rawDisk)["x86_64"].Packages = []v1.RawDiskPackage{{Name: "oci:what", Target: "what"}}
+			rawDisk = config.NewRawDisk()
+			rawDisk.X86_64.Packages = []v1.RawDiskPackage{{Name: "oci:what", Target: "what"}}
 
 			cfg.Repos = []v1.Repository{{URI: "test"}}
 		})
@@ -251,7 +251,7 @@ var _ = Describe("Runtime Actions", func() {
 			_ = fs.WriteFile(filepath.Join(partsDir, "oem.part"), []byte(""), os.ModePerm)
 			_ = fs.WriteFile(filepath.Join(partsDir, "efi.part"), []byte(""), os.ModePerm)
 
-			err := action.BuildDiskRun(cfg, rawDisk, "raw", "", "", filepath.Join(outputDir, "disk.raw"))
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "raw", "", "", filepath.Join(outputDir, "disk.raw"))
 			Expect(err).ToNot(HaveOccurred())
 			// Check that we copied all needed files to final image
 			Expect(memLog.String()).To(ContainSubstring("efi.part"))
@@ -287,7 +287,7 @@ var _ = Describe("Runtime Actions", func() {
 			_ = fs.WriteFile(filepath.Join(partsDir, "oem.part"), []byte(""), os.ModePerm)
 			_ = fs.WriteFile(filepath.Join(partsDir, "efi.part"), []byte(""), os.ModePerm)
 
-			err := action.BuildDiskRun(cfg, rawDisk, "raw", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "raw", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
 			Expect(err).ToNot(HaveOccurred())
 			// Check that we copied all needed files to final image
 			Expect(memLog.String()).To(ContainSubstring("efi.part"))
@@ -323,7 +323,7 @@ var _ = Describe("Runtime Actions", func() {
 			_ = fs.WriteFile(filepath.Join(partsDir, "oem.part"), []byte(""), os.ModePerm)
 			_ = fs.WriteFile(filepath.Join(partsDir, "efi.part"), []byte(""), os.ModePerm)
 
-			err := action.BuildDiskRun(cfg, rawDisk, "gce", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "gce", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
 			Expect(err).ToNot(HaveOccurred())
 			// Check that we copied all needed files to final image
 			Expect(memLog.String()).To(ContainSubstring("efi.part"))
@@ -356,7 +356,7 @@ var _ = Describe("Runtime Actions", func() {
 			_ = fs.WriteFile(filepath.Join(partsDir, "oem.part"), []byte(""), os.ModePerm)
 			_ = fs.WriteFile(filepath.Join(partsDir, "efi.part"), []byte(""), os.ModePerm)
 
-			err := action.BuildDiskRun(cfg, rawDisk, "azure", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "azure", "OEM", "REC", filepath.Join(outputDir, "disk.raw"))
 			Expect(err).ToNot(HaveOccurred())
 			// Check that we copied all needed files to final image
 			Expect(memLog.String()).To(ContainSubstring("efi.part"))
@@ -478,21 +478,15 @@ var _ = Describe("Runtime Actions", func() {
 			Expect(hex.EncodeToString(header.Features[:])).To(Equal("00000002"))
 			Expect(hex.EncodeToString(header.DataOffset[:])).To(Equal("ffffffffffffffff"))
 		})
-		It("Fails if the specs does not have an arch", func() {
-			rawDisk = &v1.RawDisk{}
-			err := action.BuildDiskRun(cfg, rawDisk, "raw", "OEM", "REC", "disk.raw")
-			Expect(err).To(HaveOccurred())
-			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("no values in the config for arch %s", cfg.Arch)))
-		})
 		It("Fails if the specs does not have packages", func() {
-			(*rawDisk)["x86_64"].Packages = []v1.RawDiskPackage{}
-			err := action.BuildDiskRun(cfg, rawDisk, "raw", "OEM", "REC", "disk.raw")
+			rawDisk.X86_64.Packages = []v1.RawDiskPackage{}
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "raw", "OEM", "REC", "disk.raw")
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("no packages in the config for arch %s", cfg.Arch)))
 		})
 		It("Fails if config has no repos", func() {
 			cfg.Repos = []v1.Repository{}
-			err := action.BuildDiskRun(cfg, rawDisk, "raw", "OEM", "REC", "disk.raw")
+			err := action.BuildDiskRun(cfg, rawDisk.X86_64, "raw", "OEM", "REC", "disk.raw")
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf("no repositories configured for arch %s", cfg.Arch)))
 		})

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -434,7 +434,7 @@ func NewResetSpec(cfg v1.Config) (*v1.ResetSpec, error) {
 	}, nil
 }
 
-func NewRawDisk(cfg v1.Config) *v1.RawDisk {
+func NewRawDisk() *v1.RawDisk {
 	var packages []v1.RawDiskPackage
 	defaultPackages := constants.GetBuildDiskDefaultPackages()
 
@@ -443,7 +443,8 @@ func NewRawDisk(cfg v1.Config) *v1.RawDisk {
 	}
 
 	return &v1.RawDisk{
-		cfg.Arch: &v1.RawDiskArchEntry{Packages: packages},
+		X86_64: &v1.RawDiskArchEntry{Packages: packages},
+		Arm64:  &v1.RawDiskArchEntry{Packages: packages},
 	}
 }
 

--- a/pkg/config/config_test.go
+++ b/pkg/config/config_test.go
@@ -416,8 +416,8 @@ var _ = Describe("Types", Label("types", "config"), func() {
 		})
 		Describe("RawDisk", Label("disk"), func() {
 			It("initiates a new RawDisk", func() {
-				disk := config.NewRawDisk(*c)
-				Expect(len((*disk)[c.Arch].Packages)).To(Equal(len(constants.GetBuildDiskDefaultPackages())))
+				disk := config.NewRawDisk()
+				Expect(len(disk.X86_64.Packages)).To(Equal(len(constants.GetBuildDiskDefaultPackages())))
 			})
 		})
 	})

--- a/pkg/types/v1/config.go
+++ b/pkg/types/v1/config.go
@@ -388,7 +388,7 @@ type BuildConfig struct {
 }
 
 type RawDisk struct {
-	X86_64 *RawDiskArchEntry `yaml:"x86_64,omitempty" mapstructure:"x86_64"`
+	X86_64 *RawDiskArchEntry `yaml:"x86_64,omitempty" mapstructure:"x86_64"` //nolint:revive
 	Arm64  *RawDiskArchEntry `yaml:"arm64,omitempty" mapstructure:"arm64"`
 }
 

--- a/pkg/types/v1/config.go
+++ b/pkg/types/v1/config.go
@@ -387,7 +387,10 @@ type BuildConfig struct {
 	Config `yaml:",inline" mapstructure:",squash"`
 }
 
-type RawDisk map[string]*RawDiskArchEntry
+type RawDisk struct {
+	X86_64 *RawDiskArchEntry `yaml:"x86_64,omitempty" mapstructure:"x86_64"`
+	Arm64  *RawDiskArchEntry `yaml:"arm64,omitempty" mapstructure:"arm64"`
+}
 
 // Sanitize checks the consistency of the struct, returns error
 // if unsolvable inconsistencies are found

